cdc/processor/sinkmanager/table_sink_wrapper.go

```diff
@@ -439,10 +440,9 @@ func convertRowChangedEvents(
 
 		size += e.Row.ApproximateBytes()
 
-		// This indicates that it is an update event,
-		// and after enable old value internally by default(but disable in the configuration).
-		// We need to handle the update event to be compatible with the old format.
-		if !enableOldValue && colLen != 0 && preColLen != 0 && colLen == preColLen {
+		// TiCDC always pull old value, but old value can be disabled in the configuration.
+		// in the such case, handle the update event to be compatible with the old format.
+		if e.Row.IsUpdate() && shouldSplitUpdate {
 			if shouldSplitUpdateEvent(e) {
```

Comment:
if the handle key is unchanged, csv and avro should also need to be converted into insert and delete